src/containerapp/azext_containerapp/custom.py

```diff
@@ -1100,115 +1107,94 @@ def patch_list(cmd, resource_group_name=None, managed_env=None, show_all=False):
-                reason=failed_reason))
-        else:
-            # Divide run-images into different parts by "/"
-            run_images_props = inspect_result["remote_info"]["run_images"]
-            if run_images_props is None:
-                results.append(dict(
-                    targetContainerName=inspect_result["targetContainerName"],
-                    targetContainerAppName=inspect_result["targetContainerAppName"],
-                    targetContainerAppEnvironmentName=inspect_result["targetContainerAppEnvironmentName"],
-                    targetResourceGroup=inspect_result["targetResourceGroup"],
-                    targetImageName=inspect_result["image_name"],
-                    oldRunImage=None,
-                    newRunImage=None,
-                    id=None,
-                    reason=not_based_mariner_reason))
-            else:
-                for run_images_prop in run_images_props:
-                    if run_images_prop["name"].find("mcr.microsoft.com/oryx/builder") != -1:
-                        run_images_prop = run_images_prop["name"].split(":")
-                        run_images_tag = run_images_prop[1]
-                        # Based on Mariners
-                        if run_images_tag.find('mariner') != -1:
-                            check_result = patchable_check(run_images_tag, oryx_run_img_tags, inspect_result=inspect_result)
-                            results.append(check_result)
-                        else:
-                            results.append(dict(
-                                targetContainerName=inspect_result["targetContainerName"],
-                                targetContainerAppName=inspect_result["targetContainerAppName"],
-                                targetContainerAppEnvironmentName=inspect_result["targetContainerAppEnvironmentName"],
-                                targetResourceGroup=inspect_result["targetResourceGroup"],
-                                targetImageName=inspect_result["image_name"],
-                                oldRunImage=run_images_tag,
-                                newRunImage=None,
-                                id=None,
-                                reason=failed_reason))
-                    else:
-                        # Not based on image from mcr.microsoft.com/oryx/builder
-                        results.append(dict(
-                            targetContainerAppName=inspect_result["targetContainerAppName"],
-                            targetContainerAppEnvironmentName=inspect_result["targetContainerAppEnvironmentName"],
-                            targetResourceGroup=inspect_result["targetResourceGroup"],
-                            oldRunImage=inspect_result["remote_info"]["run_images"],
-                            newRunImage=None,
-                            id=None,
-                            reason=mcr_check_reason))
+        results.append(_get_patchable_check_result(inspect_result, oryx_run_images))
```

Comment:
Should we move `_get_patchable_check_result` to `_utils` or `patch_utils` in the feature?